src/System.Management.Automation/help/ManpageHelpProvider.cs

```diff
@@ -0,0 +1,160 @@
+// Licensed under the MIT License.
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Globalization;
+using System.IO;
+using System.Reflection;
+using System.Xml;
+
+using System.Diagnostics;
+
+#nullable enable
+
+namespace System.Management.Automation
+{
+    /// <summary>
+    /// Class ProviderHelpProvider implement the help provider for manpages.
+    /// </summary>
+    internal class ManpageHelpProvider : HelpProviderWithCache
+    {
+        /// <summary>
+        /// Constructor for HelpProvider.
+        /// </summary>
+        internal ManpageHelpProvider(HelpSystem helpSystem) : base(helpSystem)
+        {
+            _sessionState = helpSystem.ExecutionContext.SessionState;
+        }
+
+        private readonly SessionState _sessionState;
+
+        #region Common Properties
+
+        /// <summary>
+        /// Name of this help provider.
+        /// </summary>
+        /// <value>Name of this help provider.</value>
+        internal override string Name
+        {
+            get
+            {
+                return "Manpage Help Provider";
+            }
+        }
+
+        /// <summary>
+        /// Help category of this provider.
+        /// </summary>
+        /// <value>Help category of this provider</value>
+        internal override HelpCategory HelpCategory
+        {
+            get
+            {
+                return HelpCategory.Manpage;
+            }
+        }
+
+        #endregion
+
+        #region Help Provider Interface
+
+        /// <summary>
+        /// Do exact match help based on the target.
+        /// </summary>
+        /// <param name="helpRequest">Help request object.</param>
+        internal override IEnumerable<HelpInfo> ExactMatchHelp(HelpRequest helpRequest)
+        {
+            foreach (ManpageInfo manpage in ManpageSearch(helpRequest.Target))
+            {
+                yield return ManpageHelpInfo.GetHelpInfo(manpage);
+            }
+        }
+
+        /// <summary>
+        /// Return manpage info that sutistfies a given target pattern one by one.
+        /// </summary>
+        /// <param name="pattern">Manpage search eacrh patter.</param>
+        public static IEnumerable<ManpageInfo> ManpageSearch(string pattern)
+        {
+            Collection<ManpageInfo> matchingManpageInfo = new Collection<ManpageInfo>();
+
+            // Get short help from commands using whatis(1)
+            try
+            {
+                using var manProc = new Process();
+                manProc.StartInfo.UseShellExecute = false;
+                manProc.StartInfo.RedirectStandardInput = false;
+                manProc.StartInfo.RedirectStandardOutput = true;
+                manProc.StartInfo.RedirectStandardError = false;
+                manProc.StartInfo.UseShellExecute = false;
+                manProc.StartInfo.CreateNoWindow = true;
+                manProc.StartInfo.FileName = "whatis";
```

Comment:
Is the utility a de-facto standard and is this present on all (Linux, MacOS) systems?

Is the parsing code below reliable or are there different dialects of the utility?

Reply:
> Is the utility a de-facto standard and is this present on all (Linux, MacOS) systems?
Is the parsing code below reliable or are there different dialects of the utility?

I don't know.  From what I found it should be standard in Linux, but I don't have a way to verify that.  Regarding MacOs, I was not able to find whether it is supported or not.  Hopped that this will become clear from the automatic tests after submitting the PR.

In any case, if `whatis` is not available, then because of the `catch` discussed [here](https://github.com/PowerShell/PowerShell/pull/15343#discussion_r657167838) PS `help` will function normally, but without showing manpage info.

I took the approach of using `whatis` and not directly accessing the manpage file since from what I read it is difficult to reliably access and parse the manpage data.  See for example [https://unix.stackexchange.com/a/418103](https://unix.stackexchange.com/a/418103).